packages/data/src/plugins/actions.js

```diff
@@ -125,6 +126,62 @@ export function* installAndActivatePlugins( plugins ) {
 	}
 }
 
+export const createErrorNotice = ( errorMessage ) => {
+	return dispatch( 'core/notices', 'createNotice', errorMessage );
+};
+
+export function* installJetpackAndConnect() {
+	try {
+		yield dispatch( STORE_NAME, 'installPlugins', [ 'jetpack' ] );
+		yield dispatch( STORE_NAME, 'activatePlugins', [ 'jetpack' ] );
+
+		const url = yield select( STORE_NAME, 'getJetpackConnectUrl', {
+			redirect_url: getAdminLink( 'admin.php?page=wc-admin' ),
+		} );
+
+		// getJetpackConnectUrl doesn't throw errors, it sets error state if
+		// something went wrong.
+		const error = yield select(
+			STORE_NAME,
+			'getPluginsError',
+			'getJetpackConnectUrl'
+		);
+
+		if ( error ) {
+			yield createErrorNotice( error );
+		} else {
+			window.location = url;
+		}
+	} catch ( error ) {
+		yield createErrorNotice( error.message );
+	}
+}
+
+export function* connectToJetpack( failureRedirect ) {
```

Comment:
It would be great to make this a control so we can just `yield connectToJetpack` inside `installJetpackAndConnect`.

Reply:
Thats a good idea. I'll do that.